src/ir/glbs.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2025 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_ir_glbs_h
+#define wasm_ir_glbs_h
+
+#include "ir/module-utils.h"
+#include "wasm.h"
+
+namespace wasm {
+
+//
+// Similar to LUBFinder, but for GLBs.
+//
+struct GLBFinder {
+  GLBFinder() {}
+
+  GLBFinder(Type initialType) { note(initialType); }
+
+  // Note another type to take into account in the GLB.
+  void note(Type type) {
+    if (type != Type::unreachable) {
+      if (glb == Type::unreachable) {
+        // This is the first thing we see.
+        glb = type;
+      } else {
+        glb = Type::getGreatestLowerBound(glb, type);
+      }
+    }
+  }
+
+  // Returns whether we noted any (reachable) value.
+  bool noted() { return glb != Type::unreachable; }
+
+  // Returns the GLB.
+  Type getGLB() { return glb; }
+
+  // Combines the information in another GLBFinder into this one, and returns
+  // whether we changed anything.
+  bool combine(const GLBFinder& other) {
+    // Check if the GLB was changed.
+    auto old = glb;
+    note(other.glb);
+    return old != glb;
+  }
+
+private:
+  // The greatest lower bound. As we go this always contains the latest value
+  // based on everything we've seen so far.
+  Type glb = Type::unreachable;
```

Comment:
We should use `Type::none` as the initial value, since it behaves very similarly to a top type whereas `Type::unreachable` is the bottom type. (This still requires some special casing, since `Type::getGLB` does not treat `Type::none` as top; we should consider fixing that to make `Type::getLUB` and `Type::getGLB` more consistent with each other.)

Reply:
Oh right, I was confused.